### PR TITLE
mgr/dashboard: Add and Delete RGW Bucket Test

### PR DESCRIPTION
**Pull request by rafaelquint**
Fixes: https://tracker.ceph.com/issues/40259?issue_count=406&issue_position=2&next_issue_id=40252&prev_issue_id=40260

Signed-off-by: Adam King adking@redhat.com
Signed-off-by: Rafael Quintero rquinter@redhat.com

### CODE REVIEW

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by p-na** [comment]
The Signed-off-by line isn't recognized because the format is still incorrect. You're missing the `<>` symbols around the email adresss.

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  beforeAll(() => {
+    page = new BucketsPage();
+  });
+
+  describe('Click Create to go to the Create page', () => {
+    beforeEach(() => {
+      page.navigateTo();
+    });
+
+    it('Should click the Create button', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]:
```suggestion
    it('should click the Create button', () => {
```

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    it('Should type name of bucket', () => {
+      // Types in the name of the bucket (which is 'testbucket'), checks we are
+      // still on the Create page
+      page.navigateTo();
+      page.getNameBox().click();
+      page.getNameBox().sendKeys('testbucket');
+      expect(Helper.getBreadcrumbText()).toEqual('Create');
+    });
+
+    it('Should set owner', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]:
```suggestion
    it('should set owner', () => {
```

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    });
+
+    it('Heading should be Create Bucket', () => {
+      // Checks if we are on the right page by checking the heading and breadcrumb
+      page.navigateTo();
+      expect(page.getCreateHeading()).toEqual('Create Bucket');
+      expect(Helper.getBreadcrumbText()).toEqual('Create');
+    });
+
+    it('Should type name of bucket', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]:
```suggestion
    it('should type name of bucket', () => {
```

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      // Sets the owner of the bucket (which is 'testid') and checks if the owner
+      // was set. Also checks if we are on the Create page
+      page.navigateTo();
+      page.getOwnerBox().click();
+      $('[value="testid"]').click();
+      expect(page.getOwnerBox().getAttribute('class')).toContain('ng-valid');
+      expect(Helper.getBreadcrumbText()).toEqual('Create');
+    });
+
+    it('Should click Create Bucket button', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]:
```suggestion
    it('should click Create Bucket button', () => {
```

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  beforeAll(() => {
+    page = new BucketsPage();
+  });
+
+  describe('Selects bucket, clicks dropdown, selects delete, checks box, and deletes ', () => {
+    beforeEach(() => {
+      page.navigateTo();
+    });
+
+    it('Should select the bucket and delete it', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]:
```suggestion
    it('should select the bucket and delete it', () => {
```

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+  beforeAll(() => {
+    page = new BucketsPage();
+  });
+
+  describe('Checks that the buckets name, owner, and count is correct', () => {
+    beforeEach(() => {
+      page.navigateTo();
+    });
+
+    it('Should check the name of the bucket is correct', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]:
```suggestion
    it('should check the name of the bucket is correct', () => {
```

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+      // testbucket is in the list
+      page.navigateTo();
+      const EC = protractor.ExpectedConditions;
+      const btable = page.getBucketTable();
+      browser.wait(EC.elementToBeClickable(btable.first()), 5000);
+      btable.getText().then(function(bucketList) {
+        expect(bucketList).toMatch('testbucket');
+      });
+    });
+    it('Should check the owner of the bucket is correct', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]:
```suggestion
    it('should check the owner of the bucket is correct', () => {
```

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+  beforeAll(() => {
+    page = new BucketsCreatePage();
+  });
+
+  describe('Types name, sets owner, and creates bucket', () => {
+    beforeEach(() => {
+      page.navigateTo();
+    });
+
+    it('Heading should be Create Bucket', () => {
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]: The `it `-sections usually start with an 'should [...]' in the beginning, could you please adapt this line as well as line [73](https://github.com/ceph/ceph/pull/28533/files#diff-c079219b5d7f8d1508461113218c595cR73)?

Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]: The `it `-sections usually start with an 'should [...]' in the beginning, could you please adapt this line as well as line [73](https://github.com/ceph/ceph/pull/28533/files#diff-c079219b5d7f8d1508461113218c595cR73)?

Comment (p-na) [reply src/pybind/mgr/dashboard/frontend/e2e/rgw/bucketCreate.e2e-spec.ts]: Proposal: `it('should have a proper heading', () => {`

That way only the test body would need to be adapted, if the heading changed.

**Comment by rafaelquint** [comment]
Made a mistake merging branches, going to make a new pull request.

**Pull request closed without merging**